library/core/src/ptr/non_null.rs

```diff
@@ -220,6 +221,7 @@ impl<T: ?Sized> NonNull<T> {
     /// ```
     #[stable(feature = "nonnull", since = "1.25.0")]
     #[rustc_const_unstable(feature = "const_nonnull_new", issue = "93235")]
+    #[must_use]
```

Comment:
Why `must_use`?

Reply:
Constructing a pointer that is immediately dropped does not achieve anything, and all odds are that it is an oversight by the developer. I can at least not think of any pattern where doing so would be preferred.

Reply:
Surely that applies to all pure functions returning a value? i.e. most `new` and `from` functions. I feel like the attribute is better reserved for when not using it is a bug, not just a mistake with no effect

Reply:
I understand your point. But looking in the standard library, other constructors such as `AtomicBool::new`, `Vec::new`, `String::new`, and `char::from_u32_unchecked` **are** in fact labelled with `#[must_use]`, even though they don't add any side effects. My point is that I don't see a clear line defining which constructors are "deserving" of this attribute. I also can't see how adding this to more constructors would do any harm (it emits, after all, just a warning).